libraries/AP_DDS/AP_DDS_Client.cpp

```diff
@@ -552,6 +555,35 @@ void AP_DDS_Client::update_topic(geographic_msgs_msg_GeoPoseStamped& msg)
 }
 #endif // AP_DDS_GEOPOSE_PUB_ENABLED
 
+#if AP_DDS_GOAL_PUB_ENABLED
+bool AP_DDS_Client::update_topic_goal(geographic_msgs_msg_GeoPointStamped& msg)
+{
+    const auto &vehicle = AP::vehicle();
+    update_topic(msg.header.stamp);
+    Location target_loc;
+    vehicle->get_target_location(target_loc);
```

Comment:
You ignore the `bool` return code here and populate the data. Check the bool, don't publish if it returns false.

Reply:
done, thanks